wallet-core/src/notes/pick.rs

```diff
@@ -40,44 +41,72 @@ pub fn notes(vk: &PhoenixViewKey, notes: NoteList, value: u64) -> NoteList {
         })
         .collect();
 
-    let sum: u64 = notes_and_values
+    // return an empty list if the sum of all notes cannot cover the cost
+    let sum: u64 = notes_values_nullifier
         .iter()
         .fold(0, |sum, &(_, value, _)| sum.saturating_add(value));
-
-    if sum < value {
+    if sum < cost {
         return NoteList::default();
     }
 
+    // if there are less that MAX_INPUT_NOTES notes, we can return the list as
+    // it is
     if notes.len() <= MAX_INPUT_NOTES {
         return notes;
     }
 
-    notes_and_values.sort_by(|(_, aval, _), (_, bval, _)| aval.cmp(bval));
-    pick_lexicographic(notes_and_values.len(), |indices| {
-        indices
-            .iter()
-            .map(|index| notes_and_values[*index].1)
-            .sum::<u64>()
-            >= value
-    })
-    .map(|index| notes_and_values[index].clone())
-    .map(|(n, _, b)| (b, n))
-    .to_vec()
-    .into()
+    // sort the input-notes from smallest to largest value
+    notes_values_nullifier.sort_by(|(_, aval, _), (_, bval, _)| aval.cmp(bval));
+
+    // return an empty list if the MAX_INPUT_NOTES highest notes do not cover
+    // the cost
+    if notes_values_nullifier
+        .iter()
+        .skip(notes_values_nullifier.len() - MAX_INPUT_NOTES)
```

Comment:
I dont understand why do we .skip here ?

Reply:
We are summing up the last 4 notes. Since at this point the notes are ordered in ascending order by their value, this will give us the max spendable.